File: Algorithm/Searching Algorithms/Binary Search/src/binary_search.cpp
/**
 * @file binary_search.cpp
 * @author Sumit Kumar (ravirajkarn@outlook.com)
 * @brief This is binary search algorithm created my me.
 * @version 0.1
 * @date 2026-05-20
 *
 * @copyright Copyright (c) 2026
 *
 */

#include <binary_search.hpp>

int main() {
  // std::vector<std::vector<int>> number = {{-1,0,3,5,9,12},{-1,0,3,5,9,12},{-1,0,5},{-1,0,3,5,9,12}};
  // std::vector<int> target = {9, 2};
  std::vector<int> number = {-1,0,3,5,9,12};
  int target = 9;
  std::print("the numbe is: {}\n ",
             /*number.size()); */ my_binary_search(number, target));
  return 0;
}

File: Algorithm/Searching Algorithms/Binary Search/src/binary_search.py
# This code is created by me
def binary_search(nums, target):
    low = 0 
    high = len(nums) - 1
    num = low + (high - low) // 2
    mid = nums[num]
    while low <= high and target >= nums[low] and target <= nums[high]:
        if mid != target:
            if mid <= target:
                low = num
            else:
                high = mid
            num = num = low + (high - low) // 2
            mid = nums[num]
            if nums[low] == target: return low
            if nums[high] == target: return high
            if low == mid: break
        else:
            return mid
    return -1
        
def main():
    number = [11, 12, 13, 14, 16, 17, 18, 19, 110, 111]
    target = 0
    print(f"the number is: {binary_search(number, target)}")

if __name__ == "__main__":
    main()